third_party/dependency_analyzer/src/main/io/bazel/rulesscala/dependencyanalyzer/ScalaVersion.scala

```diff
@@ -0,0 +1,112 @@
+package third_party.dependency_analyzer.src.main.io.bazel.rulesscala.dependencyanalyzer
+
+import scala.language.experimental.macros
+import scala.reflect.macros.blackbox
+
+object ScalaVersion {
+  val Current: ScalaVersion = ScalaVersion(util.Properties.versionNumberString)
+
+  def apply(versionString: String): ScalaVersion = {
+    versionString.split("\\.") match {
+      case Array(superMajor, major, minor) =>
+        new ScalaVersion(superMajor.toInt, major.toInt, minor.toInt)
+      case _ =>
+        throw new Exception(s"Failed to parse version $versionString")
+    }
+  }
+
+  /**
+   * Runs [code] only if minVersion and maxVersion constraints are met.
+   *
+   * NOTE: This method should be used only rarely. Most of the time
+   * just comparing versions in code should be enough. This is needed
+   * only when the code we want to run can't compile under certain
+   * versions. The reason to use this rarely is the API's inflexibility
+   * and the difficulty in debugging this code.
+   *
+   * Each of minVersion and maxVersion can either be the empty string ("")
+   * to signify that there is no restriction on this bound.
+   *
+   * Or it can be a string of a full version number such as "2.12.10".
+   *
+   * Note only literal strings are accepted, no variables etc. i.e.
```

Comment:
Added comments about otherwise not compiling. Looks like inlineable constants indeed are allowed (I guess because they get inlined before macro expansion) so added comment to that effect. If one does variables and stuff then basically that needs to be resolved at compile time which is often not possible, and even when it is (I guess a non-inlined variable which we can still analyze as having a definite value) it would be quite difficult to do so by examining the AST, AFAIK.